Library/Homebrew/test/formula_spec.rb

```diff
@@ -1996,4 +1996,28 @@ def install
       }.to raise_error("ignore_missing_libraries is available on Linux only")
     end
   end
+  
+  describe "#generate_completions_from_executable" do
+    let(:f) do
+      Class.new(Testball) do
+        def install
+          bin.mkpath
+          (bin/"foo").write <<-EOF
+            echo completion
+          EOF
+
+          FileUtils.chmod "+x", bin/"foo"
+
+          generate_completions_from_executable(bin/"foo", "test")
+        end
+      end.new
+    end
+
+    it "generates completion scripts" do
+      f.brew { f.install }
+      expect(f.bash_completion/"foo").to be_a_file
+      expect(f.zsh_completion/"_foo").to be_a_file
+      expect(f.fish_completion/"foo.fish").to be_a_file
```

Comment:
This test formula's name is `testball`, so you'll need to update these checks to look for the correct file:

```suggestion
      expect(f.bash_completion/"testball").to be_a_file
      expect(f.zsh_completion/"_ testball").to be_a_file
      expect(f.fish_completion/"testball.fish").to be_a_file
```

Or, change the `generate_completions_from_executable` call to specify the base name:

```ruby
generate_completions_from_executable(bin/"foo", "test", base_name: "foo")
```

Reply:
Thanks 🤦🏻‍♂️